packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -298,8 +300,23 @@ class WebReleaseBundle extends Target {
       if (!outputFile.parent.existsSync()) {
         outputFile.parent.createSync(recursive: true);
       }
-      inputFile.copySync(outputFile.path);
       outputResourcesFiles.add(outputFile);
+      // insert a random hash into the requests for main.dart.js and service_worker.js
+      if (environment.fileSystem.path.basename(inputFile.path) == 'index.html') {
+        final String randomHash = Random().nextInt(10000000).toString();
```

Comment:
I could also use a content hash, but technically it needs to be the content hash of the _entire_ bundle in order to be correct. Use a large random number is a pretty close approximation, because it will only change when at least one aspect of the build has changed (causing this logic to rerun)

Reply:
I think `nextInt` [accepts up to 1 << 32](https://api.dart.dev/stable/2.9.1/dart-math/Random/nextInt.html), so you could go as high as: `4294967296`. (Not sure it'd make a big difference)

Reply:
since it is embedded as a string we could go bigger, 4294967296 sounds like a nice number though!